.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,36 @@
+name: Build and Publish JAR Packages
+on:
+  workflow_dispatch:
+    inputs:
+      release_type:
+        type: choice
+        description: 'The type of release'
+        options:
+        - Major
+        - Minor
+        - Patch
+        - Snapshot
+      publish_to_maven:
+        description: 'True to publish the artifacts to Maven repository, false to skip the step'
+        default: false
+        required: false
+        type: boolean
+      java_version:
+        type: string
+        default: '11'
+      publish_vulnerabilities:
+        type: string
+        default: 'true'
+  pull_request:
+
+jobs:
+  build-and-pubish:
+    name: Build and publish JAR packages to Maven repository
+    uses: IABTechLab/uid2-shared-actions/.github/workflows/shared-java-publish-versioned-package.yaml@kcc-UID2-2327-update-maven-repos-to-use-new-pipelines
```

Comment:
will fix the version after https://github.com/IABTechLab/uid2-shared-actions/pull/47/files is merged

Reply:
Actually I can just change it to v2 now, fixed